### PR TITLE
[video_player] ExoPlayer 2.9.1, Android 9, Demo fixes

### PR DESCRIPTION
**Pull request by ened**
This upgrades the video_player dependencies on Android to the latest ExoPlayer version, 2.9.1.

Along with that, security requirements for Android 9 have been applied.

ExoPlayer 2.9.1 is now shipped in Java 8 compatible byte code. This is fully supported on modern Android toolchains, however - users will need to upgrade their build.gradle configuration to include the following section, else **their builds will break**:

```
android {
    compileOptions {
        sourceCompatibility 1.8
        targetCompatibility 1.8
    }
}
```

I consider this change inevitable and flutter should always use the latest build environments.
We might want to document it somewhere, though. Also, the plugin / project templates will need a update.

### CODE REVIEW

File: packages/video_player/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 0.7.3
```
Comment (dnfield) [inline packages/video_player/CHANGELOG.md]: This is a breaking change.  We should bump to 0.8.x

File: packages/video_player/pubspec.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: video_player
 description: Flutter plugin for displaying inline video with other Flutter
   widgets on Android and iOS.
 author: Flutter Team <flutter-dev@googlegroups.com>
-version: 0.7.2
+version: 0.7.3
```
Comment (dnfield) [inline packages/video_player/pubspec.yaml]: ditto on version

File: packages/video_player/example/android/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.2-all.zip
```
Comment (dnfield) [inline packages/video_player/example/android/gradle/wrapper/gradle-wrapper.properties]: Do we need to bump this here? 

I would probably prefer to keep all the gradle versions in sync for this repo - if we need to upgrade it here, could we move it to a separate PR and bump all the gradle-wrapper.properties files?

**Comment by dnfield** [comment]
Thanks for doing this!

A few minor adjustments/requests, but I think this is a good idea overall!

File: packages/video_player/example/android/gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (ened) [reply packages/video_player/example/android/gradle/wrapper/gradle-wrapper.properties]: Alright.. I was surprised to find 4.10 in use, as 4.10.2 has been released soon after. Will reactivate #817 and send asap.

**Comment by ened** [comment]
@dnfield Thank you for reviewing. 

Here are the details rgd. the Java 8 language change:
https://developer.android.com/studio/write/java8-support

File: packages/video_player/example/android/gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (dnfield) [reply packages/video_player/example/android/gradle/wrapper/gradle-wrapper.properties]: I'm fine with going to 4.10.2, but I'd just like to do it for all of them at once so that it's easier to find/replace on the whole repo in the future - and preferably do it as a separate commit so we can track the history and revert if need be.

Comment (ened) [reply packages/video_player/example/android/gradle/wrapper/gradle-wrapper.properties]: OK. Updated #817 ..

**Comment by dnfield** [comment]
I think the Java 8 stuff should be fine, we're not using Jack or retrolambda.

If you want to update any of the syntax for Java 8 that should be fine, as long as we don't end up using any language features that break earlier version support.

This LGTM, thanks for the contribution!
